pkg/forwarder/forwarder.go

```diff
@@ -163,20 +179,26 @@ func New(cfg Config) (*Worker, error) {
 		if !pool.AppendCertsFromPEM(data) {
 			level.Warn(logger).Log("msg", "no certs found in from-ca-file")
 		}
-		fromTransport.TLSClientConfig.RootCAs = pool
+
+		fromTransport.TLSClientConfig = &tls.Config{
+			Certificates: []tls.Certificate{cert},
+			RootCAs:      pool,
+		}
```

Comment:
It would be better to set only the fields that we specifically want to update rather than replacing the full struct.

```suggestion

		fromTransport.TLSClientConfig.Certificates = []tls.Certificate{cert}
		fromTransport.TLSClientConfig.RootCAs = pool
```